java/java/JavaParser.g4

```diff
@@ -254,7 +254,7 @@ arrayInitializer
     ;
 
 classOrInterfaceType
-    : typeIdentifier typeArguments? ('.' identifier typeArguments?)*
+    : (identifier typeArguments? '.')* typeIdentifier typeArguments?
```

Comment:
Looks good. Annotations? https://docs.oracle.com/javase/specs/jls/se19/html/jls-4.html#jls-ClassOrInterfaceType

Reply:
Hmm, this was closed before I could respond, but this is covered in the typearguments no?
Anyway, let's check it up when it comes up next time...

```
typeArguments
    : '<' typeArgument (',' typeArgument)* '>'
    ;

typeArgument
    : typeType
    | annotation* '?' ((EXTENDS | SUPER) typeType)?
    ;

typeType
    : annotation* (classOrInterfaceType | primitiveType) (annotation* '[' ']')*
    ;
```